src/mlpack/core/data/save_image.cpp

```diff
@@ -54,6 +54,11 @@ bool SaveImage(const std::string& filename,
     Log::Warn << "Only the first image will be saved!" << std::endl;
   }
 
+    if (info.Width() * info.Height() * info.Channels() != image.n_elem)
+  {
+    Log::Fatal << "The dimension given does not match the image dimesion. \n";
```

Comment:
```suggestion
    Log::Fatal << "data::Save(): The given image dimensions do not match the "
        << "dimensions of the matrix to be saved!" << std::endl;    
```

I think it could be useful to be a little more explicit here with the error message; what do you think?